services/minespace-web/src/components/pages/Tailings/TailingsSummaryPage.js

```diff
@@ -161,31 +164,23 @@ export const TailingsSummaryPage = (props) => {
           </Col>
         </Row>
         <Divider />
-        <Tabs
-          tabPosition="left"
-          activeKey={activeTab}
-          defaultActiveKey={tabs[0]}
-          onChange={(tab) => handleTabChange(tab)}
-          className="vertical-tabs"
-        >
-          {tabs.map((tab) => {
-            return (
-              <Tabs.TabPane
-                tab={formatUrlToUpperCaseString(tab)}
-                disabled={disabledTabs}
-                key={tab}
-                className="vertical-tabs--tabpane"
-              >
-                <TailingsSummaryForm
-                  mineGuid={mineGuid}
-                  handleSaveData={handleSaveData}
-                  handleTabChange={handleTabChange}
-                  initialValues={initialValues}
-                />
-              </Tabs.TabPane>
-            );
-          })}
-        </Tabs>
+        <SteppedForm
+          tabs={tabs}
+          initialValues={initialValues}
+          handleSaveData={handleSaveData}
+          handleTabChange={handleTabChange}
+          handleSaveDraft={handleSaveData}
+          errors={errors}
+          match={match}
+          form={
```

Comment:
This looks pretty clean! Thoughts on modelling a bit more on the `<Tab` component? So you would pass in the tabs as subcomponents themselves. That way the conditional rendering of the different steps could also be internal to the `SteppedForm` component

```js
<SteppedForm
          currentStep={tab}
          initialValues={initialValues}
          handleSaveData={handleSaveData}
          handleTabChange={handleTabChange}
          handleSaveDraft={handleSaveData}
          errors={errors}
          match={match}>
    <Step key="basic-information">
          <BasicInformation initialValues={initialValues} />
    </Step>
    <Step key="engineer-of-record">
             <EngineerOfRecord initialValues={initialValues} /
     </Step>
    ...
</SteppedForm>
```

Reply:
That's a good idea.  I had been keeping passing in the form, but it's been stripped of any specific usefulness at this point so we can easily wrap this current component in the redux form wrapper and pass things in this way.